### PR TITLE
Add support for Bolt 5.4 an API telemetry

### PR DESCRIPTION
**Pull request by robsdedude**
Starting with Bolt 5.4, the driver will, by default, send anonymous API usage statistics to the server if requested.

The driver configuration `telemetry_disabled=True` can be used to disable this.

```python
import neo4j

with neo4j.GraphDatabase.driver(..., telemetry_disabled=True) as driver:
    ...
```

The driver transmits the following information:

* Every time one of the following APIs is used to execute a query (for the first time), the server is informed of this (without any further information like arguments, client identifiers, etc.):
  * `driver.execute_query`
  * `session.begin_transaction`
  * `session.execute_read`, `session.execute_write`
  * `session.run`
  * the async counterparts of the above methods

Depends on:
 * https://github.com/neo4j-drivers/testkit/pull/594

### CODE REVIEW

File: src/neo4j/_async/io/_bolt5.py [hunk 1]
```diff
@@ -665,3 +671,22 @@ def get_base_headers(self):
...
+
+class AsyncBolt5x4(AsyncBolt5x3):
+
+    PROTOCOL_VERSION = Version(5, 4)
+
+    def telemetry(self, api: TelemetryAPI, dehydration_hooks=None,
+                  hydration_hooks=None, **handlers) -> None:
+        if (
+            self.telemetry_disabled
+            or not self.configuration_hints.get("telemetry.enabled", False)
```
Comment (StephenCathcart) [inline src/neo4j/_async/io/_bolt5.py:684]: Is self.configuration_hints configured/returned from the server?

Comment (robsdedude) [reply src/neo4j/_async/io/_bolt5.py:684]: Yes. All configuration hints that come from the server are being stored on the connection object on `SUCCESS`ful `HELLO` https://github.com/neo4j/neo4j-python-driver/blob/fe85b713853edea0566c691aaebc6fd66c6ea9eb/src/neo4j/_async/io/_bolt5.py#L135

File: tests/unit/async_/io/test_class_bolt.py [hunk 2]
```diff
@@ -65,7 +65,8 @@ def test_class_method_protocol_handlers():
         ((5, 1), 1),
         ((5, 2), 1),
         ((5, 3), 1),
-        ((5, 4), 0),
+        ((5, 4), 1),
+        ((5, 5), 0),
```
Comment (StephenCathcart) [inline tests/unit/async_/io/test_class_bolt.py:69]: How come we need 5.5 here? Is this just future-proofing?

File: src/neo4j/_async/io/_bolt5.py (hunk 1 above)
Comment (StephenCathcart) [reply src/neo4j/_async/io/_bolt5.py:684]: Thanks for clarifying!

**Review by StephenCathcart — APPROVED** [review]
Looks good to me! :shipit:

File: tests/unit/async_/io/test_class_bolt.py (hunk 2 above)
Comment (robsdedude) [reply tests/unit/async_/io/test_class_bolt.py:69]: It asserts that there are `0` Bolt 5.5 handlers registered, i.e., the driver does *not* support Bolt 5.5.

Comment (StephenCathcart) [reply tests/unit/async_/io/test_class_bolt.py:69]: Ah ha that makes more sense, thanks.

**Pull request merged**